introduction_to_amazon_algorithms/xgboost_abalone/xgboost_inferenece_script_mode.ipynb

```diff
@@ -0,0 +1,448 @@
+    "2. The IAM role arn used to give training and hosting access to your data. See the documentation for how to create these. Note, if more than one role is required for notebook instances, training, and/or hosting, please replace the boto regex with a the appropriate full IAM role arn string(s)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "import io\n",
+    "import os\n",
+    "import boto3\n",
+    "import sagemaker\n",
+    "import time\n",
+    "import urllib\n",
+    "\n",
+    "role = sagemaker.get_execution_role()\n",
+    "region = boto3.Session().region_name\n",
+    "\n",
+    "# S3 bucket for saving code and model artifacts.\n",
+    "# Feel free to specify a different bucket here if you wish.\n",
+    "bucket = sagemaker.Session().default_bucket()\n",
+    "prefix = 'sagemaker/DEMO-xgboost-inference-script-mode'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fetching the dataset\n",
+    "\n",
+    "Following methods download the Abalone dataset and upload files to S3."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "# Load the dataset\n",
+    "FILE_DATA = 'abalone'\n",
+    "urllib.request.urlretrieve(\"https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/abalone\", FILE_DATA)\n",
+    "sagemaker.Session().upload_data(FILE_DATA, bucket=bucket, key_prefix=prefix+'/train')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training the XGBoost model\n",
+    "\n",
+    "SageMaker can now run an XGboost script using the XGBoost estimator. A typical training script loads data from the input channels, configures training with hyperparameters, trains a model, and saves a model to `model_dir` so that it can be hosted later. In this notebook, we use the same training script [abalone.py](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/introduction_to_amazon_algorithms/xgboost_abalone/abalone.py) from [Regression with Amazon SageMaker XGBoost algorithm](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/introduction_to_amazon_algorithms/xgboost_abalone/xgboost_abalone_dist_script_mode.ipynb). Refer to [Regression with Amazon SageMaker XGBoost algorithm](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/introduction_to_amazon_algorithms/xgboost_abalone/xgboost_abalone_dist_script_mode.ipynb) for details on the training script.\n",
+    "\n",
+    "After setting training parameters, we kick off training, and poll for status until training is completed, which in this example, takes between few minutes.\n",
+    "\n",
+    "To run our training script on SageMaker, we construct a sagemaker.xgboost.estimator.XGBoost estimator, which accepts several constructor arguments:\n",
```

Comment:
Enclose the name of the estimator class in backticks.

Reply:
Because this notebook is demonstrating the inference script, maybe add a cell with:

`! pygementize inference.py`